exporter/datadogexporter/traces_exporter_test.go

```diff
@@ -0,0 +1,303 @@
+			Labels: map[string]string{
+				"B": "C",
+			},
+		},
+		Spans: []*tracepb.Span{
+			{
+				TraceId:                 []byte{0x01},
+				SpanId:                  []byte{0x02},
+				Name:                    &tracepb.TruncatableString{Value: "root"},
+				Kind:                    tracepb.Span_SERVER,
+				SameProcessAsParentSpan: &wrapperspb.BoolValue{Value: true},
+				Attributes: &tracepb.Span_Attributes{
+					AttributeMap: map[string]*tracepb.AttributeValue{
+						"span_attr_name": {
+							Value: &tracepb.AttributeValue_StringValue{
+								StringValue: &tracepb.TruncatableString{Value: "Span Attribute"},
+							},
+						},
+					},
+				},
+				Resource: &resourcepb.Resource{
+					Type: "override",
+					Labels: map[string]string{
+						"B": "D",
+					},
+				},
+				TimeEvents: &tracepb.Span_TimeEvents{
+					TimeEvent: []*tracepb.Span_TimeEvent{
+						{
+							Time: &timestamppb.Timestamp{
+								Seconds: 0,
+								Nanos:   0,
+							},
+							Value: &tracepb.Span_TimeEvent_Annotation_{
+								Annotation: &tracepb.Span_TimeEvent_Annotation{
+									Description: &tracepb.TruncatableString{Value: "Some Description"},
+									Attributes: &tracepb.Span_Attributes{
+										AttributeMap: map[string]*tracepb.AttributeValue{
+											"attribute_name": {
+												Value: &tracepb.AttributeValue_StringValue{
+													StringValue: &tracepb.TruncatableString{Value: "Hello MessageEvent"},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			{
+				TraceId:                 []byte{0x01},
+				SpanId:                  []byte{0x03},
+				ParentSpanId:            []byte{0x02},
+				Name:                    &tracepb.TruncatableString{Value: "client"},
+				Kind:                    tracepb.Span_CLIENT,
+				SameProcessAsParentSpan: &wrapperspb.BoolValue{Value: true},
+				Links: &tracepb.Span_Links{
+					Link: []*tracepb.Span_Link{
+						{
+							TraceId: []byte{0x04},
+							SpanId:  []byte{0x05},
+							Type:    tracepb.Span_Link_PARENT_LINKED_SPAN,
+							Attributes: &tracepb.Span_Attributes{
+								AttributeMap: map[string]*tracepb.AttributeValue{
+									"span_link_attr": {
+										Value: &tracepb.AttributeValue_IntValue{
+											IntValue: 12345,
+										},
+									},
+								},
+							},
+						},
+					},
+					DroppedLinksCount: 0,
+				},
+			},
+			{
+				TraceId:                 []byte{0x01},
+				SpanId:                  []byte{0x04},
+				ParentSpanId:            []byte{0x03},
+				Name:                    &tracepb.TruncatableString{Value: "server"},
+				Kind:                    tracepb.Span_SERVER,
+				SameProcessAsParentSpan: &wrapperspb.BoolValue{Value: false},
+			},
+		},
+	}
+
+	// ensure that the protobuf serialized traces payload contains HostName Env and Traces
+	// ensure that the json gzipped stats payload contains HostName Env and Stats
+	got := testTraceExporterHelper(internaldata.OCToTraceData(td), t)
```

Comment:
I think we really shouldn't be using `internaldata` anymore.

Reply:
Gotcha, no worries, just was using it to generate some mock data like the other vendor exporter tests, but can update these mocks when i get a sec to use the preferred format.

Reply:
So i spent some time looking into this. This would be trivial to mock properly without the OC helper funcs, using something like: `testdata.GenerateTraceDataTwoSpansSameResourceOneDifferent()` except that, If i understand correctly, `"go.opentelemetry.io/collector/internal/data/testdata"` is not accessible for the contrib repo since it's an internal package to opentelemetry-collector.

So I'm quite stuck here, as I don't see any other examples of how to mock this info in tests without a helper func, no other exporters have to do this in contrib without relying on OC helers, and the exporter's that handle it in core have those internal helper funcs. Wouldn't it just be easier if `go.opentelemetry.io/collector/internal/data/testdata` were accessible to contrib tests?

Reply:
Moving to internal is something relatively new. What I've been doing is to create a helper func close to where it's used, like this:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a4b57fc84b100c3a5e852360a26be18be5cdfecb/processor/groupbytraceprocessor/processor_test.go#L771-L788

Reply:
makes sense, i've gone ahead and used a helper along these lines and removed the OC stuff